File: CoreService/Models/Practice.cs
using System;
using System.Collections.Generic;

namespace CoreService.Models;

public partial class Practice
{
    public int Id { get; set; }

    public int Studentid { get; set; }

    public int Themeid { get; set; }

    public string? Finalgrade { get; set; }

    public string Status { get; set; } = null!;

    public DateTime Createddate { get; set; }

    public DateTime Updateddate { get; set; }

    public virtual Student Student { get; set; } = null!;

    public virtual Theme Theme { get; set; } = null!;
}

File: frontend/src/entities/Practice.ts
import { Student } from "./Student.ts";
import { Theme } from "./Theme.ts";
import { Consultant } from "./Consultant.ts";
import { Lecturer } from "./Lecturer.ts";

export interface Practice {
    id: number;
    studentid: number;
    themeid: 0;
    type: string;
    finalgrade: string;
    status: string;
    createddate: Date;
    updateddate: Date;
    student: Student;
    theme: Theme;
    supervisor: Lecturer;
    consultant: Consultant;
}

export interface InputPractice {
    studentid: number;
    themeid: number;
    type: string;
    finalgrade: string;
    status: string;
}